libnetdata/dictionary/dictionary.h

```diff
@@ -202,4 +227,22 @@ extern size_t dictionary_stats_walkthroughs(DICTIONARY *dict);
 
 extern int dictionary_unittest(size_t entries);
 
+// ----------------------------------------------------------------------------
+// STRING implementation
+
+typedef struct string STRING;
```

Comment:
I have a concern to use only `STRING` as a name, and I think we should use something  like `NETDATA_STRING`, because contributors can have confusion with data type [string](https://en.cppreference.com/w/cpp/header/string) in c++ libraries.  A good example for source of confusion is Microsof behavior to rename datat ypes in [upper case](https://docs.microsoft.com/en-us/windows/win32/learnwin32/working-with-strings)